libpod/image/pull.go

```diff
@@ -356,11 +420,36 @@ func checkRemoteImageForLabel(ctx context.Context, label string, imageInfo pullR
 	if err != nil {
 		return err
 	}
+	return isLabelInInspect(label, remoteInspect.Labels, imageInfo.image)
+}
+
+// checkRemoteImageIsUpToDate checks if the remote image has a specific label. If not, an error is returned.
+// Then it checks if an source reference is newer than lastUpdated, or the time an image we are checking was last updated.
+// It returns false if lastUpdate is older than the source image, true otherwise
+func checkRemoteImageIsUpToDate(ctx context.Context, label *string, lastUpdated time.Time, imageInfo pullRefPair, sc *types.SystemContext) (bool, error) {
+	labelImage, err := imageInfo.srcRef.NewImage(ctx, sc)
+	if err != nil {
+		return false, err
+	}
+	remoteInspect, err := labelImage.Inspect(ctx)
+	if err != nil {
+		return false, err
+	}
+	if label != nil {
+		if err = isLabelInInspect(*label, remoteInspect.Labels, imageInfo.image); err != nil {
+			return false, err
+		}
+	}
+	return !remoteInspect.Created.After(lastUpdated), nil
```

Comment:
- `types.ImageInspectInfo.Created` may be `nil`; this code needs to handle that… somehow.
- (@rhatdan, others) Do we _actually_ want to prefer the local image when it has a later creation timestamp, anyway? It seems to me that whatever the registry says right now is “the source of truth” (i.e. if the registry reverts to an earlier image, so should we) and we should just always pull the remote image if it’s not exactly the same (same config and/or same manifest). (**Note:** Changing this would also require changing the documentation.)

And, _if_ we end up with “the registry is always right”, that gets us _very_ close to “always pull (assuming redundant pulls are trivially cheap), but ignore an error if the destination exists”.  See also the other note about the use of `image.Names`.

Reply:
Can you point out the specific point in Docker where the decision is made, please?

Reply:
I am going to revise my previous statement and say there is no where in docker this behavior happens. It was only asked for in BZ. In my research I did find a pull request for the change (1498 in docker/cli) but it hasn't been merged. Thus, I don't know where the user in the BZ got the inclination this behavior exists.

Reply:
It’s _fine_ to define an entirely new behavior if we think it’s going to be useful, but if the whole point of this PR is to be feature-compatible with Docker, then we really need to start with being clear about what the Docker behavior that we want to emulate is.

OTOH, if the option name was changed, we are not going to be 100% feature-compatible anyway, so… let’s go back to where this feature request came from.